modules/social_features/social_event/modules/social_event_max_enroll/src/Controller/UserEnrollInviteControllerAlter.php

```diff
@@ -0,0 +1,95 @@
+<?php
+
+namespace Drupal\social_event_max_enroll\Controller;
+
+use Drupal\Core\Session\AccountProxyInterface;
+use Drupal\Core\Url;
+use Drupal\node\NodeInterface;
+use Drupal\social_event\EventEnrollmentInterface;
+use Drupal\social_event_invite\Controller\UserEnrollInviteController;
+use Drupal\social_event_max_enroll\Service\EventMaxEnrollService;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use Symfony\Component\HttpFoundation\RedirectResponse;
+use Symfony\Component\HttpFoundation\RequestStack;
+
+/**
+ * Accepts or declines an event enrollment invite.
+ *
+ * @package Drupal\social_event_invite\Controller
+ */
+class UserEnrollInviteControllerAlter extends UserEnrollInviteController {
+
+  /**
+   * The event maximum enroll service.
+   *
+   * @var \Drupal\social_event_max_enroll\Service\EventMaxEnrollService
+   */
+  protected EventMaxEnrollService $eventMaxEnrollService;
+
+  /**
+   * {@inheritdoc}
+   */
+  public function __construct(
+    RequestStack $requestStack,
+    AccountProxyInterface $currentUser,
+    EventMaxEnrollService $eventMaxEnrollService
+  ) {
+    parent::__construct($requestStack, $currentUser);
+
+    $this->eventMaxEnrollService = $eventMaxEnrollService;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container): self {
+    return new static(
+      $container->get('request_stack'),
+      $container->get('current_user'),
+      $container->get('social_event_max_enroll.service')
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function updateEnrollmentInvite(EventEnrollmentInterface $event_enrollment, string $accept_decline): RedirectResponse {
+    // We should move forward only when the user is accepting the invite.
+    if ($accept_decline === '1') {
+      $event_max_enroll_service = $this->eventMaxEnrollService;
+
+      /** @var \Drupal\social_event\Entity\EventEnrollment $event_enrollment */
```

Comment:
Is the @var tag really necessary when the parameter has type defined?

Reply:
Yes, since without it we will have **PHPStan** error:
```
 ------ -------------------------------------------------------------------------------------------- 
  Line   UserEnrollInviteControllerAlter.php                                                         
 ------ -------------------------------------------------------------------------------------------- 
  61     Call to an undefined method Drupal\social_event\EventEnrollmentInterface::getFieldValue().  
 ------ -------------------------------------------------------------------------------------------- 
```

Reply:
I have updated it.